tools/astarte_export/lib/fetchdata/fetchdata.ex

```diff
@@ -0,0 +1,398 @@
+      endpoint_id = mapping.endpoint_id
+      path = mapping.endpoint
+      data_type = mapping.value_type
+      data_field = get_data_field_name(data_type)
+
+      {:ok, result} =
+        Queries.retrive_individual_datastreams(
+          conn,
+          realm,
+          device_id,
+          interface_id,
+          endpoint_id,
+          path,
+          data_field
+        )
+
+      values =
+        Enum.to_list(result)
+        |> Enum.map(fn map ->
+          atom_data_field = String.to_atom(data_field)
+          return_value = map[atom_data_field]
+          value = from_native_type(return_value, data_type)
+
+          reception_timestamp =
+            map[:reception_timestamp]
+            |> DateTime.from_unix!(:millisecond)
+            |> DateTime.to_iso8601()
+
+          Map.new()
+          |> Map.put(:value, value)
+          |> Map.put(:reception_timestamp, reception_timestamp)
+        end)
+
+      case values do
+        [] ->
+          acc1
+
+        _ ->
+          values1 =
+            Map.new()
+            |> Map.put(:path, path)
+            |> Map.put(:value, values)
+
+          [values1 | acc1]
+      end
+    end)
+  end
+
+  def fetch_object_datastreams(conn, realm, [h | _] = mappings, device_id, storage) do
+    fullpath = h.endpoint
+    [_, endpointprefix, _] = String.split(fullpath, "/")
+    path = "/" <> endpointprefix
+
+    extract_2nd_level_params =
+      Enum.reduce(mappings, [], fn mapping, acc1 ->
+        path = mapping.endpoint
+        [_, _, suffix] = String.split(path, "/")
+        data_type = mapping.value_type
+        [%{suffix_path: suffix, data_type: data_type} | acc1]
+      end)
+
+    {:ok, result} = Queries.retrive_object_datastream_value(conn, realm, storage, device_id, path)
+
+    values =
+      Enum.to_list(result)
+      |> Enum.map(fn map ->
+        reception_timestamp =
+          map[:reception_timestamp]
+          |> DateTime.from_unix!(:millisecond)
+          |> DateTime.to_iso8601()
+
+        list = Map.to_list(map)
+
+        value_list =
+          List.foldl(list, [], fn {key, value}, acc ->
+            case to_string(key) do
+              "v_" <> item ->
+                match_object =
+                  Enum.find(extract_2nd_level_params, fn map1 -> map1[:suffix_path] == item end)
+
+                case match_object do
+                  nil ->
+                    acc
+
+                  _ ->
+                    data_type = match_object[:data_type]
+                    token = "/" <> match_object[:suffix_path]
+                    value1 = from_native_type(value, :double)
+
+                    case value1 do
+                      "" ->
+                        acc
+
+                      _ ->
+                        [%{:name => token, :value => value1} | acc]
+                    end
+                end
+
+              _Other ->
+                acc
+            end
+          end)
+
+        Map.new()
```

Comment:
Just build the final map instead of adding one key at a time:
```
%{
  reception_timestamp: reception_timestamp,
  path: path,
  value: value
}
```